core/src/main/scala/org/apache/spark/deploy/master/MasterSource.scala

```diff
@@ -42,6 +42,6 @@ private[spark] class MasterSource(val master: Master) extends Source {
 
   // Gauge for waiting application numbers in cluster
   metricRegistry.register(MetricRegistry.name("waitingApps"), new Gauge[Int] {
-    override def getValue: Int = master.waitingApps.size
+    override def getValue: Int = master.apps.filter(_.state == ApplicationState.WAITING).size
```

Comment:
@jerryshao In all my tests using multiple Spark versions, the "apps" and "waitingApps" metrics have been the same. Do you know if the "waitingApps" metric is intended to refer to something other than the number of apps that are currently in the waiting state?
Thanks